hooks/useToolChat.tsx

```diff
@@ -65,18 +64,50 @@ const useToolChat = (question?: string, toolName?: any) => {
         content: question as string,
         role: "user",
       });
-      setIsGeneratingReport(true);
-      const { reportContent, rawContent } = await getFullReport(tiktokAnalysis);
-      setTiktokReportContent(reportContent);
-      setTiktokRawReportContent(rawContent);
-      setIsGeneratingReport(false);
       initReport();
       setBeginCall(false);
     };
     if (!beginCall || !question) return;
     init();
   }, [beginCall, question]);
 
+  useEffect(() => {
+    const save = async () => {
+      const uniqueId = `${address}-${Date.now()}`;
+      const response = await saveReportReferenece(
+        uniqueId,
+        messages[1].content,
+        tiktokRawReportContent,
+        tiktokNextSteps,
+      );
+      if (response?.error) return;
+      const referenceId = response.id;
+      await finalCallback(
+        {
+          id: uuidV4(),
+          content: "TikTok Report",
+          role: "assistant",
+        },
+        {
+          id: uuidV4(),
+          content: question as string,
+          role: "user",
+        },
+        conversationId as string,
+        referenceId,
+        uniqueId,
+      );
+    };
+    if (
+      !loading &&
+      messages?.length === 2 &&
+      tiktokRawReportContent &&
+      tiktokNextSteps &&
+      toolName === Tools.getSegmentsReport
+    )
+      save();
+  }, [loading, messages, tiktokNextSteps, tiktokRawReportContent]);
+
```

Comment:
File too long

## **Development Environment**

### **Tools**

- Task Management: Linear
- Version Control: Github

### **Code Standards**

- Maximum file length: 99 lines
- Directory Structure:
    - `/components`: HTML code only
    - `/libs`: Helper functions and API calls
    - `/hooks`: State management and helper function calls
    - `/providers`: Hooks with state shared across multiple components

- requirements: https://zany-hourglass-531.notion.site/Developer-Requirements-Document-15026a7d835c8062bbcde49b59aa00e8